toxcore/tox.h

```diff
@@ -800,10 +805,52 @@ void tox_iterate(Tox *tox);
 
 
 /**
+ * Multiple nospam values can be used to have different addresses at the same time.
+ * This can help to block friend request spam for one address
+ * leaving other addresses valid.
+ * Also this can help to distinguish friend requests by nospam values they used.
+ *
+ * Every non zero 32-bit unsigned integer can be used as nospam value.
+ *
+ * It's also possible to block all friend requests by
+ * removing all nospam values from nospam set.
+ *
+ * Each nospam value can have a description, that can help
+ * user to distinguish friend requests.
+ */
+
+typedef enum TOX_ERR_NOSPAM {
+    TOX_ERR_NOSPAM_OK,
+
+    /**
+     * Nospam number is not in nospam set.
+     */
+    TOX_ERR_NOSPAM_NOT_FOUND,
+
+    /**
+     * Impossible to add new nospam value to nospam set.
+     */
+    TOX_ERR_NOSPAM_TOO_MANY,
```

Comment:
8 Why don't you use `MALLOC` error other functions provide to indicate a similar failure?